### PR TITLE
Task icon row

### PR DESCRIPTION
**Pull request by CaseyHillers**
When you long tap the icon, it will show the task name in a Tooltip.

There was some left over formatting from #459 in the tests folder.

image

I tried adding tests to ensure Images were loading for the correct stage name, but couldn't find a way to differentiate Images from each other.

### CODE REVIEW

**Comment by jonahwilliams** [comment]
Does this work with a mouse?

File: app_flutter/lib/status_grid.dart [hunk 1]
```diff
@@ -69,14 +70,33 @@ class StatusGrid extends StatelessWidget {
             gridDelegate: SliverGridDelegateWithFixedCrossAxisCount(
                 crossAxisCount: columnCount),
             itemBuilder: (BuildContext context, int gridIndex) {
-              final int statusIndex = gridIndex ~/ columnCount;
+              if (gridIndex == 0) {
+                /// The top left corner of the grid is nothing since
+                /// the left column is for [CommitBox] and the top
+                /// row is for [TaskIcon].
+                return Container();
```
Comment (jonahwilliams) [inline app_flutter/lib/status_grid.dart]: use a `const SizedBox()` instead

File: app_flutter/lib/task_icon.dart [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+  static const String stageCirrus = 'cirrus';
+  static const String stageLuci = 'chromebot';
+  static const String stageDevicelab = 'devicelab';
+  static const String stageDevicelabWin = 'devicelab_win';
+  static const String stageDevicelabIOs = 'devicelab_ios';
+
+  /// A lookup table for matching [stageName] to [Image].
+  ///
+  /// [stageName] is based on the backend.
+  static Map<String, Image> stageIcons = <String, Image>{
```
Comment (jonahwilliams) [inline app_flutter/lib/task_icon.dart]: Should be final

File: app_flutter/lib/task_icon.dart [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+  @override
+  Widget build(BuildContext context) {
+    return Tooltip(
+      message: task.name,
+      child: Container(
+        margin: const EdgeInsets.all(7.5),
+        child: stageIcons.containsKey(task.stageName)
+            ? stageIcons[task.stageName]
+            : Icon(Icons.help),
```
Comment (jonahwilliams) [inline app_flutter/lib/task_icon.dart]: `Icon(Icons.help)` should be const, right? I'm surprised that the lints didn't catch this one

File: app_flutter/lib/status_grid.dart (hunk 1 above)
Comment (CaseyHillers) [reply app_flutter/lib/status_grid.dart]: Done

File: app_flutter/lib/task_icon.dart (hunk 3 above)
Comment (CaseyHillers) [reply app_flutter/lib/task_icon.dart]: Yes, linter didn't catch it. Added.

**Comment by CaseyHillers** [comment]
@jonahwilliams it works with a mouse on Android emulator. I assumed using Tooltip from the framework would work the same on the web, but right now the project does not build for web.

See #450

File: app_flutter/lib/task_icon.dart (hunk 2 above)
Comment (CaseyHillers) [reply app_flutter/lib/task_icon.dart]: Can it be made final? I get issues from the constructor saying it can't be final since Image.asset is not const.

Comment (jonahwilliams) [reply app_flutter/lib/task_icon.dart]: `static final Map<String, Image> stageIcons = ...` should be fine

Comment (CaseyHillers) [reply app_flutter/lib/task_icon.dart]: Thanks, that fixed it

**Comment by jonahwilliams** [comment]
The web builds should work again once we roll build_runner

**Pull request merged**